src/Workspaces/Core/Portable/Serialization/SerializerService.cs

```diff
@@ -55,9 +55,10 @@ private protected SerializerService(SolutionServices workspaceServices)
     {
         _workspaceServices = workspaceServices;
 
-        // Serialization is only involved when we have a remote process.  Which is only in VS. So the type of the
-        // storage service here is well known.
-        _storageService = (TemporaryStorageService)workspaceServices.GetRequiredService<ITemporaryStorageServiceInternal>();
+        // Serialization to temporary storage is only involved when we have a remote process.  Which is only in VS. So the type of the
+        // storage service here is well known.  However the serializer is created in other cases (e.g. to compute project state checksums).
+        // So lazily instantiate the storage service to avoid attempting to get the TemporaryStorageService when not available.
+        _storageService = new Lazy<TemporaryStorageService>(() => (TemporaryStorageService)workspaceServices.GetRequiredService<ITemporaryStorageServiceInternal>());
```

Comment:
A bit weird.  the serializer service is almost only used for oop serialization - except its also used for create project parse option checksums on all platforms.

The TemporaryStorageService is not used in that case, so instantiating it lazily to avoid throwing when it attempts to convert to TemporaryStorageService